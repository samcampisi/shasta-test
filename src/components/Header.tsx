import React from 'react';
import { StyleProp, ViewStyle, Text, TextStyle, SafeAreaView } from 'react-native';
import styles from '../styles/Header.style';

export interface HeaderProps {
  title: string;
  style?: StyleProp<ViewStyle>;
  textStyle?: StyleProp<TextStyle>;
  testID?: string;
}

const Header = (props: HeaderProps) => {
  const { title, style, textStyle, testID } = props;

  return (
    <SafeAreaView testID={testID} style={[styles.header, style]}>
      <Text style={[styles.headerText, textStyle]}>{title}</Text>
    </SafeAreaView>
  );
};

export default Header;
