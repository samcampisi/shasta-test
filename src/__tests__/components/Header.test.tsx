import Header from '../../components/Header';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { render } from 'react-native-testing-library';

Enzyme.configure({ adapter: new Adapter() });

const createTestProps = (props?: object) => ({
  ...props,
});

describe('Header', () => {
  const props: any = createTestProps({
    title: 'Test title',
  });

  describe('Basic button', () => {
    const { getByText, toJSON } = render(<Header {...props} />);

    it('should render the title', () => {
      expect(getByText('Test title')).toBeDefined();
    });

    it('should match snapshot', () => {
      expect(toJSON()).toMatchSnapshot();
    });
  });
});
