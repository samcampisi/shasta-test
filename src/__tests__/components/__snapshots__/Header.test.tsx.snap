// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Basic button should match snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "alignItems": "center",
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 16,
          "marginVertical": 10,
        },
        undefined,
      ]
    }
  >
    Test title
  </Text>
</RCTSafeAreaView>
`;
